### ISSUE TITLE
Unable to navigate make's errors/warnings in the PROBLEMS window

### ISSUE THREAD

**Issue by TryerGit** [microsoft/vscode-makefile-tools#211]
So far, I have used `make` (on a linux machine) via `tasks.json` in a "raw" fashion and without any extension thus:

        "tasks": [
            {
                "label": "dbgbuild",
                "type": "shell",
                "command": "make",
                "args": [
                    "CONF=Debug",
                    "-C",
                    "./.vscode"
                ],
                "group": "build",
                "problemMatcher": {
                    "owner": "cpp",
                    "fileLocation": [
                        "relative",
                        "${workspaceFolder}/.vscode"
                    ],
                    "pattern": {
                        "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
                        "file": 1,
                        "line": 2,
                        "column": 3,
                        "severity": 4,
                        "message": 5
                    }
                }
            },
            {
                "label": "releasebuild",
                "type": "shell",
                "command": "make",
                "args": [
                    "CONF=Release",
                    "-C",
                    "./.vscode"
                ],
                "group": "build",
                "problemMatcher": {
                    "owner": "cpp",
                    "fileLocation": [
                        "relative",
                        "${workspaceFolder}/.vscode"
                    ],
                    "pattern": {
                        "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
                        "file": 1,
                        "line": 2,
                        "column": 3,
                        "severity": 4,
                        "message": 5
                    }
                }
            }
        ]

The above defines two tasks: `dbgbuild` which executes `make CONF=Debug -C ./.vscode` and `releasebuild` which executes `make CONF=Release -C ./.vscode`

As you would know, the `problemMatcher` contains information to correctly capture the warnings and errors that the `make` process gives rise to and shows these errors/warnings in the `PROBLEMS` window in VSCode.

In trying to achieve the above using the extension, following the documentation, in `settings.json`, I have 

    "makefile.configurations": [
        {
            "name": "dbgbuild",
            "makeArgs": ["CONF=Debug"]
        },
        {
            "name": "releasebuild",
            "makeArgs": ["CONF=Release"]
        }
    ],
    "makefile.makeDirectory": "./.vscode"

However, there does not seem to be any equivalent of `problemMatcher` in this extension that would correctly translate `make`'s errors and warnings into a list that can be navigated in the `PROBLEMS` window in VSCode. Indeed, on using the extension to build my target, while the errors and warnings are displayed in the `OUTPUT` window as raw text, they do not appear in the `PROBLEMS` window in a manner in which the user can navigate from error to error.

How can this functionality be achieved via the extension?

**Comment by andreeis** [comment]
@TryerGit, problem matcher support is scheduled for the next release (2-3 months out).
But you can use the tasks.json as you wrote it until then, in parallel with the rest of the features that are available in the extension (except the build command). Let us know if there is something not working with Makefile Tools and your tasks.json.

Closing as a duplicate of https://github.com/microsoft/vscode-makefile-tools/issues/7.

**Comment by andreeis** [comment]
@TryerGit, building a target with Makefile Tools will have problem matchers support in the 0.3 release so you will be able to see errors/warnings in the Problems tab without a tasks.json workaround. While we wrap-up the release, you can test a vsix from [this link](https://github.com/microsoft/vscode-makefile-tools/actions/runs/1468495870) (extensions tab --> context menu of "..." in the upper right corner --> install from vsix).

Unfortunately, due to a VSCode limitation, you can only reference a problem matcher that is supported by VSCode or any installed extensions via a string name (like "$gcc"). We opened a feature request to the VSCode team to expand the tasks APIs to allow specifying a problem matcher with all the fields, as opposed to a string only.

Makefile Tools and CppTools define the most usual $gcc pattern.
In your .vscode/settings.json, define "problemMatchers" string array for the entries of your "makefile.configurations" array, like in the below example.

    "makefile.configurations": [
        {
            "name": "MyConfiguration",
            "problemMatchers": ["$gcc", "$unknown"],
            "makeArgs": ["V=1"]
        },
    ],

Is there anything different than $gcc that you would need? Until VSCode improves the tasks and problem matcher APIs, the only way you get a new problem matcher is by defining it in the package.json of the Makefile Tools extension and an update on Market Place. We can help with that if you let us know what other regular expressions you need, besides $gcc.

**Comment by TryerGit** [comment]
Andreeis:

At present, on Linux I only use gcc. So this is fine and I will test it in due course. On Windows, I do not use makefiles anyway. There, I have a tasks.json that calls MSBuild.exe.

**Comment by bobbrow** [comment]
Just as FYI, vscode defines a problem matcher that I believe works with MSBuild/MSVC called `$msCompile`

**Comment by TryerGit** [comment]
> Just as FYI, vscode defines a problem matcher that I believe works with MSBuild/MSVC called `$msCompile`

Yes, I am aware of that. I have had some trouble with that on Windows as well.

See https://github.com/microsoft/vscode/issues/131611

It works on and off.

**Comment by andreeis** [comment]
@TryerGit, we released Makefile Tools 0.3.0 today which includes support for problem matchers. Let us know if you encounter any other issues.

If you would like, you can also leave us a review on [Market Place](https://marketplace.visualstudio.com/items?itemName=ms-vscode.makefile-tools&ssr=false#review-details).
